### PR TITLE
LOG-1282: Generate Dockerfile from Dockerfile.in

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR
* Adds script to generate Dockerfile from midstream Dockerfile.in
* Adds linting tasks to check dockerfile and code linting
* Fixes issues related to code linting

cc @alanconway @vimalk78 

### Links
https://issues.redhat.com/browse/LOG-1282

### CODE REVIEW

**Review by alanconway — APPROVED** [review]
/lgtm

File: Dockerfile [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
-COPY version ./version
-COPY scripts ./scripts
-COPY files ./files
-COPY go.mod .
-COPY go.sum .
-COPY vendor ./vendor
-COPY manifests ./manifests
-COPY .bingo .bingo
-COPY Makefile ./Makefile
-COPY pkg ./pkg
```
Comment (vimalk78) [inline Dockerfile:16]: suggest to keep these

**Pull request merged**